_posts/2015-09-04-releasecop-tracks-stale-releases.markdown

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "Releasecop Tracks Stale Releases"
+date: 2015-09-01 17:30
+comments: true
+author: Joey Aghion
+github-url: https://www.github.com/joeyAghion
+twitter-url: https://twitter.com/joeyAghion
+blog-url: http://joey.aghion.com
+categories: [Ruby, open source, OSS, testing, Continuous Integration, Continuous Deployment, devops, tools]
+---
+
+Artsy practices a sort of [continuous delivery](http://en.wikipedia.org/wiki/Continuous_delivery). We keep release cycles short and the process of reviewing, testing, and deploying our software as reliable, fast, and automated as possible. (This blog has touched on these practices [multiple](http://artsy.github.io/blog/categories/testing/) [times](http://artsy.github.io/blog/categories/continuous-integration).)
+
+Usually, commits that have been reviewed and merged are immediately built and tested. Successfully built versions of the codebase are often automatically deployed to a staging environment. On an automated or frequent-but-manual basis, that version is deployed to a production environment. Thus, commits form a pipeline:
+
+* From developers' working branches
+* To the master branch
+* Through a hopefully-successful build
+* To a staging environment
+* To production
+
+The number of apps and services we deploy has grown to _dozens_ per team, so sometimes things fall through the cracks. We've been using [Releasecop](https://github.com/joeyAghion/releasecop) for the last few months to get gentle email reminders when an environment could use a deploy.
+
+<!-- more -->
+
+    gem install releasecop
+    releasecop edit
+
+This opens a _manifest_ file where you can describe the sequence of git remotes and branches that make up your own release pipeline. For example:
```

Comment:
I'd rather not link to the real manifest since it enumerates private repos (I know, I know... lots we can improve there). This sample includes the both the simplest and most complex examples.